### ISSUE TITLE
`clippy::let_underscore_lock` false-positive when binding mutex without locking

### ISSUE THREAD

**Issue by daprilik** [rust-lang/rust-clippy#8486]
### Summary

`let_underscore_lock` should only fire if the `.lock()` method is being invoked (and/or the type being bound is a lock guard)

### Lint Name

let_underscore_lock

### Reproducer

I tried this code:

```rust
pub fn main() {
    let x = parking_lot::Mutex::new(0);
    let _ = x;
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
error: non-binding let on a synchronization lock
 --> src/main.rs:3:5
  |
3 |     let _ = x;
  |     ^^^^^^^^^^
  |
  = note: `#[deny(clippy::let_underscore_lock)]` on by default
  = help: consider using an underscore-prefixed named binding or dropping explicitly with `std::mem::drop`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_lock
```

I expected to see this happen:

Nothing

### Version

```text
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: x86_64-pc-windows-msvc
release: 1.59.0
LLVM version: 13.0.0
```

**Comment by daprilik** [comment]
Scanning through the code, the issue seems to be caused by these lines:

https://github.com/rust-lang/rust-clippy/blob/a98e7ab8b94485be6bd03e0c6b8682ecab5b52e6/clippy_lints/src/let_underscore.rs#L112-L113

These are paths should be pointing at the lock _guards_, rather than the locks themselves.
